elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java

```diff
@@ -86,4 +87,16 @@ default void flush() {
         // default to ignoring criteria
         return loadObjects(entityClass);
     }
+
+    /**
+     * Filter a collection by the Predicates in filterScope.
+     *
+     * @param collection the collection to filter
+     * @param type
+     * @param filterScope the FilterScope containing the set of Predicates
+     * @return the filtered collection
+     */
+    default Collection filterCollection(Collection collection, String type, FilterScope<?> filterScope) {
```

Comment:
I'm not 100% the concept of FilterScope really makes sense here.  All we really need is a list<Predicate> - not a scope.  The scope concept was more around checks.  I'm not a huge fan of the catchall 'scope' class where we stick a bunch of unrelated stuff.